newIDE/app/src/MainFrame/Editors/ExternalLayoutEditor.js

```diff
@@ -88,6 +91,16 @@ export default class ExternalLayoutEditor extends BaseEditor {
     });
   };
 
+  saveUiSettings = () => {
+    const layout = this.getExternalLayout();
+    if (layout) {
```

Comment:
already tried that, still error. I found that doing this works:
const uiSettings = this.editor ? this.editor.getUiSettings() : {}

Reply:
actually does the order matter? In my failed attempt it was `if ( this.editor && layout)`

Reply:
Order doesn't matter as you're not calling a function and both conditions are not linked together.
What was the error with `if (this.editor && layout)`.
It does not make sense to create an empty object, it could be actually quite dangerous to feed an empty object into unserializeFromJSObject, because the C++ side might be expecting so properties.
In practice it won't crash. In theory, you should avoid it: adding `{}` is just a hack here ;)

Reply:
ah I see, lets try again. I think the flow was failing because I was using directly this.editor, instead of putting it in a constant